src/kvstore/kvstore_dist.h

```diff
@@ -147,8 +147,11 @@ class KVStoreDist : public KVStoreLocal {
       for (const auto& v : values) {
         v.WaitToWrite();
       }
+      // wait for rank 0
+      Barrier();
     } else {
-      // do nothing
+      // wait for rank 0
+      Barrier();
     }
     if (!ps::Postoffice::Get()->is_recovery()) {
```

Comment:
The `Barrier()` call at line 157 should be sufficient?

Reply:
1. I put 3 worker and 3 ps in a single machine. The Barrier() seems doesn't work. 
The wrong mesage is "[11:44:42] src/kvstore/././kvstore_dist_server.h:422: Check failed: !stored.is_none() init 82 first".
2. Then I try to put another Barrier() in 158, like below. The problem no longer occurs. 
```
if (!ps::Postoffice::Get()->is_recovery()) {
        std::cout<<keys[0]<<"-------------barrier-----------"<<ps::kWorkerGroup<<std::endl;
        Barrier();
        Barrier();
}
```

Why only one Barrier() seems doesn't work? 
@eric-haibin-lin